pkg/controllers/provisioning/scheduling/topologydomaingroup.go

```diff
@@ -22,51 +22,76 @@ import (
 	"sigs.k8s.io/karpenter/pkg/scheduling"
 )
 
+type topologyDomain struct {
```

Comment:
I think we should call this something like `topologyDomainAttributes`, the topology domain is the string in the original map (e.g. `us-west-2a` would be a domain for a `topology.kubernetes.io/zone` topology).